### PR TITLE
Add a title to the link node to support titles in markdown

### PR DESCRIPTION
**Pull request by themagickoala**
We have a use case for supporting titles in link markdown nodes (see [markdown spec](https://www.markdownguide.org/basic-syntax/#adding-titles "Even github does it!")). We're currently copying source code and modifying it inside our own project, but that's not really sustainable so here's a PR to add it to Lexical.

Note that this doesn't add support for adding the title via the WYSIWYG editor, but it will support adding it in Markdown and then converting back to rich text.

### CODE REVIEW

File: packages/lexical-react/src/LexicalTreeView.tsx [hunk 1]
```diff
@@ -604,6 +608,15 @@ function printRelProperties(node: LinkNode) {
   return str;
 }
 
+function printTitleProperties(node: LinkNode) {
+  let str = node.getTitle();
+  // TODO Fix nullish on LinkNode
+  if (str != null) {
```
Comment (zurfyx) [inline packages/lexical-react/src/LexicalTreeView.tsx:614]: If I'm reading this right, it's guaranteed to be `null` or `string` no? So we can do `!== null` instead and drop the TODO?

**Review by zurfyx — APPROVED** [review]
LGTM, thank you! cc @fantactuka if you want to double check the MD stuff

File: packages/lexical-react/src/LexicalTreeView.tsx (hunk 1 above)
Comment (themagickoala) [reply packages/lexical-react/src/LexicalTreeView.tsx:614]: Probably, but this was a copy and paste of target/rel versions, which are also string | null.

File: packages/lexical-markdown/src/MarkdownTransformers.ts [hunk 2]
```diff
@@ -340,11 +343,11 @@ export const LINK: TextMatchTransformer = {
       return linkContent;
     }
   },
-  importRegExp: /(?:\[([^[]+)\])(?:\(([^()]+)\))/,
-  regExp: /(?:\[([^[]+)\])(?:\(([^()]+)\))$/,
+  importRegExp: /(?:\[([^[]+)\])(?:\((?:([^()\s]+)(?:\s"([^"]+)"\s*)?)\))/,
+  regExp: /(?:\[([^[]+)\])(?:\((?:([^()\s]+)(?:\s"([^"]+)"\s*)?)\))$/,
```
Comment (fantactuka) [inline packages/lexical-markdown/src/MarkdownTransformers.ts:347]: Any chance to check if " was escaped in title?

```
[Text](https://test.com "Title \" with escaped quote")
```

**Review by fantactuka — APPROVED** [review]

File: packages/lexical-markdown/src/MarkdownTransformers.ts (hunk 2 above)
Comment (themagickoala) [reply packages/lexical-markdown/src/MarkdownTransformers.ts:347]: Good catch, I'll try and add a test case for that too, and fix the other unit test that's broken.

**Comment by themagickoala** [comment]
@zurfyx can this be merged now?

**Comment by themagickoala** [comment]
@fantactuka @zurfyx I've rebased this to fix the conflict created by my other PR (#4168) so it should be good to merge now I think.

File: packages/lexical-react/src/LexicalTreeView.tsx (hunk 1 above)
Comment (acywatson) [reply packages/lexical-react/src/LexicalTreeView.tsx:614]: I'm fine with this - we can fix it now or we can merge this and fix them all in a separate PR -all good either way for me.

**Comment by zurfyx** [comment]
Thank you! Sorry for the delay

**Pull request merged**